### ISSUE TITLE
rewrite attester mock to use `mockapi.attesterConfig = LEGACY_GLOBAL_ATTESTER_MOCK_CONFIG` 

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-browser#5021]
> **Second wave of PRs**: allow local configuration of the mock APIs, but give every test the same configuration:
> 
> I would recommend to do this service by service, eg start with only Attester, then FES in another PR, etc.
> 
> ```ts
>   ava.default('some test', testWithBrowser(undefined, assignedPort, async (t, browser) => { 
>     const mockApi = new MockApi();
>     mockApi.fesConfig = LEGACY_GLOBAL_FES_MOCK_CONFIG;
>     mockApi.ekmConfig = LEGACY_GLOBAL_EKM_MOCK_CONFIG
>     mockApi.attesterConfig = LEGACY_GLOBAL_ATTESTER_MOCK_CONFIG;
>     mockApi.googleConfig = LEGACY_GLOBAL_GOOGLE_MOCK_CONFIG;
>     await mockApi.withMockedApis(assignedPort, async () => {
>       // run browser tests as before
> ```
> 
> This way, we ensure that all current schenanigans we do for tests can be configured using the new mechanism. But we don't yet do individual configuration per test, because it's a task on its own to figure out which tests need exactly what configuration.
>

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-browser/issues/4494#issuecomment-1471550025_

**Comment by tomholub** [comment]
From @sosnovsky comment:

> Yes, currently it's not exposed, I was planning to add it to `AvaContext`:
> ```ts
> t.mockApi = await startMockApiAndCopyBuild(t);
> ```
> (here - https://github.com/FlowCrypt/flowcrypt-browser/blob/master/test/source/test.ts#L68)
> 
> And then use it in tests like this:
> ```ts
> t.mockApi.fesConfig = { ... };
> t.mockApi.ekmConfig = { ... };
> ```
> 

So we'll start with Attester since that is a very straight forward to configure, small mock. I guess it will be something like this? A particular test example:

```diff
    test(
      'no.pub@client-configurations-test - no backup, no keygen, enforce attester submit with submit err',
      testWithBrowser(undefined, async (t, browser) => {
+       t.mockApi = await startMockApiAndCopyBuild(t);
+       t.mockApi.attesterConfig = LEGACY_GLOBAL_ATTESTER_MOCK_CONFIG;
        const settingsPage = await BrowserRecipe.openSettingsLoginApprove(t, browser, 'no.pub@client-configuration-test.flowcrypt.test');
        await SetupPageRecipe.manualEnter(
          settingsPage,
          'no.pub.client.configuration',
          { noPrvCreateClientConfiguration: true, enforceAttesterSubmitClientConfiguration: true, fillOnly: true },
          { isSavePassphraseChecked: false, isSavePassphraseHidden: false }
        );
        await settingsPage.waitAndClick('@input-step2bmanualenter-save');
        await settingsPage.waitAll(['@container-overlay-prompt-text', '@action-overlay-retry']);
        const renderedErr = await settingsPage.read('@container-overlay-prompt-text');
        expect(renderedErr).to.contain(`Attempting to import unknown key`);
        expect(renderedErr).to.contain(
          `Your organization requires public keys to be present on company LDAP server, but no public key was found for no.pub@client-configuration-test.flowcrypt.test. Please ask your internal help desk.`
        );
      })
    );
```

And a bunch more tests like that.

Then `LEGACY_GLOBAL_ATTESTER_MOCK_CONFIG` will be, for now, set once as:

```ts
const LEGACY_GLOBAL_ATTESTER_MOCK_CONFIG = {
  pubkeyLookup: {
    'mock.only.pubkey@flowcrypt.com': somePubkey,
    'test.match.attester.key@gmail.com': testMatchPubKey,
    'test.mismatch.attester.key@gmail.com': somePubkey,
    // ... and so on
  },
  ldapRelay: {
    // similar structure
  },
  welcomeMessageEnabled?: boolean 
}
```

Then large chunks from `mockAttesterEndpoints` will be removed, so that it doesn't contain data, but just uses configuration.

Once we see the PR, then we can discuss more details and steps after that so that we can agree on the exact approach. Of course, after that, the next PR will dis-entangle that, so that each test configures its own specific Attester configuration instead of using the `LEGACY_GLOBAL_ATTESTER_MOCK_CONFIG`, which can then be removed.

If you prefer to do both steps in one PR because it's less work, you can. I imagine it will be easier to manage it in two separate PRs.

Once Attester is done, then we'll go through the other easy mocks (EKM, WKD, backend, keysOpenPGPOrg, SKS) then the medium difficulty endpoints (FES 2x) then the most difficult one (Google).

**Comment by ioanmo226** [comment]
@sosnovsky, I am going to implement this task starting from tomorrow.
Please let me know if you have other tasks that you believe are higher priority and would like me to take a look at.

**Comment by sosnovsky** [comment]
This one is a good choice for next task, should improve our test's structure 👍
